postgresql/client_test.go

```diff
@@ -140,3 +118,59 @@ func TestWriteCommand(t *testing.T) {
 		t.Fatal("Wrong cnt: ", cnt)
 	}
 }
+
+func TestPgAdvisoryLock(t *testing.T) {
+	db := dbSetup(t)
+	lock, err := util.NewPgAdvisoryLock(1, db)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !lock.Locked() {
+		t.Error("Couldn't obtain the lock")
+	}
+
```

Comment:
Actually, now that I'm looking at this again, I think it's fine to leave out the comment I suggested. This logic tests that the PgAdvisoryLock works on the Postgres-side, which is a valid micro-test.

Instead, I would be happier if there were a test that instantiates an election and tests it with two "clients" / requests (even though these requests might be sequential). Does that make sense? Or maybe I am once again misunderstanding what's going on in Go.
It looks like you test the RestElection below, but what about the regular election you had implemented?